### PR TITLE
OOP school library: Create Classes

### PR DESCRIPTION
**Pull request by iamsjunaid**
# Pull Request Summary:

### Description:
In this pull request, I have created separate classes for `person`, `student`, and `teacher`. The changes are summarized as follows:

1. `person` Class:
   - The `person` class has been refactored and moved into a new file named `person.rb`.
   - The class now includes instance variables `@id`, `@name`, and `@age`, along with corresponding getters and setters for each of them.
   - The constructor has been updated to include optional parameters for `name` and `parent_permission`, with default values of "Unknown" and `true`, respectively.
   - A private method `generate_id` has been implemented to assign a random ID to each person object, ensuring uniqueness.
   - The `can_use_services?` method has been modified to check if the person is 18 years or older or has parent permission, returning `true` or `false` accordingly.

2. `student` Class:
   - A new `student` class has been created in the file `student.rb`, inheriting from the `person` class.
   - The `student` class extends the parent's constructor by adding the `@classroom` instance variable and a parameter to set it during object creation.
   - The class includes a new method `play_hooky`, which returns the fun response "¯\(ツ)/¯" when invoked.
   - The `can_use_services?` method in the `student` class has been overridden to behave similarly to the `person` class, allowing students to use services if they are 18 years or older or have parent permission.

3. `teacher` Class:
   - A new `teacher` class has been implemented in the `teacher.rb` file, also inheriting from the `person` class.
   - The `teacher` class extends the parent's constructor by adding the `@specialization` instance variable and a parameter to set it during object creation.
   - The `can_use_services?` method in the `teacher` class has been overridden to always return `true`, granting teachers unrestricted access to services.

These changes significantly improve the code's organization, maintainability, and reusability. The modular design allows for better separation of concerns, making it easier to manage and extend the codebase. Please review this pull request at your earliest convenience. Your feedback is highly valued and will contribute to the project's improvement. Thank you.

Best regards,
Junaid

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class Person
+  attr_reader :id, :name, :age
```
Comment (DeliceLydia) [inline person.rb]: - [ ] I suggest that you make `name` and `age` attributes accessor because we need to read and modify them

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @iamsjunaid, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Created required classes
- No linter errors


## Required Changes ♻️
_Check the comments under the review._



### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by iamsjunaid** [comment]
Hi @DeliceLydia. Thanks for the thorough review😊. I have made the requested changes.

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @iamsjunaid,

✔️ Implementation of classes in different files

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**